compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2028,7 +2028,12 @@ where
         TyAndLayout { ty: this.ty, layout }
     }
 
-    fn field(this: TyAndLayout<'tcx>, cx: &C, i: usize) -> C::TyAndLayout {
+    fn ty_and_layout_kind(
```

Comment:
This can just be a helper method defined inside the field method

Reply:
Good point!